test/test_torch.py

```diff
@@ -9581,6 +9581,44 @@ def test_lerp(self, device):
                 expected = start + weight * (end - start)
                 self.assertEqual(expected, actual)
 
+    def _test_logaddexp(self, device, dtype, base2=False):
+        if base2:
+            gt_func = np.logaddexp2
+            our_func = torch.logaddexp2
+        else:
+            gt_func = np.logaddexp
+            our_func = torch.logaddexp
+
+        a = torch.randn(51, 2, dtype=dtype, device=device)
```

Comment:
This is good but to test vectorized and non-vectorized code paths you want to test with one array of at least 64 contiguous elements and another of fewer than four elements:

```
def _helper(vals_a, vals_b):
  a = torch.tensor(vals_a, device=device, dtype=dtype)
  b = torch.tensor(vals_b, device=device, dtype=dtype)
  
  actual = fn(a, b)
  expected = torch.from_numpy(ref(a.cpu().numpy(), b.cpu().numpy()))
  
  self.assertEqual(actual, expected)

vals_a = list((random.random() - .5 for _ in range(64)))
vals_b = list((random.random() - .5 for _ in range(64)))
_helper(vals_a, vals_b)
_helper(vals_a[:4], vals_b[:4])

vals_a = list(((random.random() - .5) * 1000 for _ in range(64)))
vals_b = list(((random.random() - .5) * 1000 for _ in range(64)))
_helper(vals_a, vals_b)
_helper(vals_a[:4], vals_b[:4])

# Tests nonfinite values
vals_a = [float('inf'), float('-inf'), float('inf'), float('nan')]
vals_b = [(float('inf'), float('-inf'), float('-inf'), float('nan')]
_helper(vals_a, vals_b)
```